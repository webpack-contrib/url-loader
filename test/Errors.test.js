/* eslint-disable
  prefer-destructuring,
*/
import webpack from '@webpack-contrib/test-utils';

describe('Errors', () => {
  test('Validation Error', async () => {
    const config = {
      loader: {
        test: /\.png$/,
        options: {
          limit: {},
        },
      },
    };

    const stats = await webpack('fixture.js', config);
    const { source } = stats.toJson().modules[0];

    // eslint-disable-next-line
    const fn = () => eval(source);

    expect(fn).toThrowErrorMatchingSnapshot();
  });
});
