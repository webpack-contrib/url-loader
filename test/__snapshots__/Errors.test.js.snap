// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Errors Validation Error 1`] = `
"Module build failed: URL Loader

  [4mOptions Validation Error[24m

  [2moptions[22m.limit  [34mshould be string,number[39m


  [2moptions[22m.limit  [34mshould be string,number[39m
    at validate (<PROJECT_ROOT>/node_modules/@webpack-contrib/schema-utils/dist/validate-options.js:87:15)
    at validator (<PROJECT_ROOT>/node_modules/@webpack-contrib/schema-utils/dist/validate-options.js:118:10)
    at Object.loader (<PROJECT_ROOT>/src/index.js:19:3)"
`;
